test/algorithms/set_operations/intersection/intersection_integer.cpp

```diff
@@ -0,0 +1,122 @@
+// Boost.Geometry
+// Unit Test
+
+// Copyright (c) 2007-2023 Barend Gehrels, Amsterdam, the Netherlands.
+
+// This file was modified by Oracle on 2015-2022.
+// Modifications copyright (c) 2015-2022, Oracle and/or its affiliates.
+// Contributed and/or modified by Menelaos Karavelas, on behalf of Oracle
+// Contributed and/or modified by Adam Wulkiewicz, on behalf of Oracle
+
+// Use, modification and distribution is subject to the Boost Software License,
+// Version 1.0. (See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+
+#include "test_intersection.hpp"
+#include <algorithms/test_overlay.hpp>
+
+#include <algorithms/overlay/overlay_cases.hpp>
+
+#define TEST_INTERSECTION(caseid, clips, points, area) \
+    (test_one<Polygon, Polygon, Polygon>) \
+    ( #caseid, caseid[0], caseid[1], clips, points, area, settings)
+
+namespace
+{
+    std::string rectangles[2] =
+    {
+        "POLYGON((1000 1000,2000 1000,2000 2000,1000 2000))",
+        "POLYGON((500 500,1500 500,1500 1500,500 1500))"
+    };
+
+    std::string start_within[2] =
+    {
+        "POLYGON((1000 1000,2000 1000,2000 2000,1000 2000))",
+        "POLYGON((1250 1250,1500 750,1750 1250,1750 1750))"
+    };
+}
+
+
+template <typename Polygon>
+void test_areal()
+{
+    static bool is_open = bg::closure<Polygon>::value == bg::open;
+
+    ut_settings settings;
+    settings.test_point_count = true;
```

Comment:
This is the way, in these tests, to verify if the closing point is removed.
Some test code (for example `test_point_count`) in this PR relates to that.